data/bills.js

```diff
@@ -0,0 +1,128 @@
+const mongoCollections = require("../config/mongoCollections");
+const bills = mongoCollections.bills;
+const users = require("./users");
+const uuid = require('node-uuid');
+
+let exportedMethods = {
+
+    getAllBills() {
+        return bill().then((billCollection) => {
+            return billCollection.find({}).toArray();
+        });
+    },
+
+    getBillById(id) {
+        return bill().then((billCollection) => {
+            return billCollection
+                .findOne({_id: id})
+                .then((bill) => {
+                    if (!bill) 
+                        throw "Bill not found";
+                    return bill;
+                });
+        });
+    },
+
+    getBillByUserId(id) {
+        return bill().then((billCollection) => {
+            return billCollection
+                .find({"userId":  id})
+                .toArray();
+        });
+    },
+
+    getBillByDate(date) {
+        return bill().then((billCollection) => {
+            return billCollection
+                .find({"date":  date})
+                .toArray();
+        });
+    },
+
+     getBillByCategory(category) {
+        return bill().then((billCollection) => {
+            return billCollection
+                .find({"category":  category})
+                .toArray();
+        });
+    },
+
+    //category and date could get from select on webpage
+    addBill(category, amount, date, note, userID ) {

```

Comment:
That would mean more work... what if the person has a budget in INR and adds a bill in USD... getting latest exchange rates is going to be a nightmare :|